services/src/main/java/com/linkedin/kmf/services/TopicRebalancer.java

```diff
@@ -0,0 +1,427 @@
+    for (Broker broker : brokers) {
+      brokersAsInt.$plus$eq(broker.id());
+    }
+
+    scala.collection.Map<Object, Seq<Object>> partitionToReplicas =
+        AdminUtils.assignReplicasToBrokers(brokersAsInt, partitionCount, _replicationFactor, 0, 0);
+
+    String jsonReassignmentData = scalaReassignmentToJson(partitionToReplicas, partitionCount);
+
+    LOG.debug(_serviceName + ": reassignments " + jsonReassignmentData + ".");
+    _zkUtils.createPersistentPath(ZkUtils.ReassignPartitionsPath().toString(), jsonReassignmentData, _zkUtils.DefaultAcls());
+
+  }
+
+  /**
+   * This should look something like
+   * <pre>
+   *   {"version":1,"partitions":[
+   *     {"topic":"kmf-topic","partition":5,"replicas":[1]},
+   *     {"topic":"kmf-topic","partition":4,"replicas":[0]},
+   *     {"topic":"kmf-topic","partition":3,"replicas":[2]},
+   *     {"topic":"kmf-topic","partition":1,"replicas":[0]},
+   *     {"topic":"kmf-topic","partition":2,"replicas":[1]},
+   *     {"topic":"kmf-topic","partition":0,"replicas":[2]}]}
+   * </pre>
+   * @return a json string
+   */
+  private String scalaReassignmentToJson(scala.collection.Map<Object, Seq<Object>> scalaPartitionToReplicas, int partitionCount) {
+    //TODO: Is this the best way to construct this JSON data structure?
+    StringBuilder bldr = new StringBuilder();
+    bldr.append("{\"version\":1,\"partitions\":[\n");
+    for (int partition = 0; partition < partitionCount; partition++) {
+      bldr.append("  {\"topic\":\"").append(_topic).append("\",\"partition\":").append(partition).append(",\"replicas\":[");
+      scala.collection.Seq<Object> replicas = scalaPartitionToReplicas.apply(partition);
+      for (int replicaIndex = 0; replicaIndex < replicas.size(); replicaIndex++) {
+        Object replica = replicas.apply(replicaIndex);
+        bldr.append(replica).append(",");
+      }
+      bldr.setLength(bldr.length() - 1);
+      bldr.append("]},\n");
+    }
+    bldr.setLength(bldr.length() - 2);
+    bldr.append("]}");
+    return bldr.toString();
+  }
+
+  /**
+   * This runs the preferred replica election for all partitions.
+   */
+  protected void runPreferredElection(ZkUtils zkUtils, List<PartitionInfo> partitionInfoList) {
+    scala.collection.mutable.HashSet<TopicAndPartition> scalaPartitionInfoSet = new scala.collection.mutable.HashSet<>();
+    for (PartitionInfo javaPartitionInfo : partitionInfoList) {
+      scalaPartitionInfoSet.add(new TopicAndPartition(_topic, javaPartitionInfo.partition()));
+    }
+    PreferredReplicaLeaderElectionCommand.writePreferredReplicaElectionData(zkUtils, scalaPartitionInfoSet);
+  }
+
+  protected void addPartitions(ZkUtils zkUtils, int addPartitionCount) {
+    AdminUtils.addPartitions(zkUtils, _topic, addPartitionCount, null, false);
+  }
+
+  /**
+   * @return the state of the topic e.g. all the partitions and replicas
+   */
+  private TopicState topicState() {
```

Comment:
Why use "get"?  It does not mean anything unless you need to comply with Java Beans.  Scala does not use it.  Java itself has moved away from it.   For example in java collections it's "size()" not "getSize()".